src/test/ui/const-generics/invalid-enum.stderr

```diff
@@ -25,70 +25,130 @@ LL |   let _: Example<CompileFlag::A, _> = Example { x: 0 };
+error[E0107]: wrong number of const arguments for `Example`: expected 1, found 0
   --> $DIR/invalid-enum.rs:36:10
    |
+LL | pub struct Example<const CF: CompileFlag, T=u32>{
+   | ------------------------------------------------ defined here, with 1 const parameter: `T`
+...
 LL |   let _: Example<Example::ASSOC_FLAG, _> = Example { x: 0 };
-   |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected 1 const argument
+   |          ^^^^^^^ ---------------------- found no const argument in here
+   |
+help: add missing const argument:
+   |
+LL |   let _: Example<Example::ASSOC_FLAG, _, T> = Example { x: 0 };
+   |                                        ^^^
 
-error[E0107]: wrong number of type arguments: expected at most 1, found 2
+error[E0107]: wrong number of type arguments for `Example`: expected at most 1, found 2
   --> $DIR/invalid-enum.rs:36:10
    |
+LL | pub struct Example<const CF: CompileFlag, T=u32>{
+   | ------------------------------------------------ defined here, with at most 1 type parameter: `CF`
+...
 LL |   let _: Example<Example::ASSOC_FLAG, _> = Example { x: 0 };
-   |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected at most 1 type argument
+   |          ^^^^^^^
+   |
+help: consider removing this type argument
    |
-help: If this generic argument was intended as a const parameter, try surrounding it with braces:
+LL |   let _: Example<Example::ASSOC_FLAG> = Example { x: 0 };
+   |                                    --
+help: if this argument was intended to be a const parameter, try surrounding it with braces:
    |
 LL |   let _: Example<{ Example::ASSOC_FLAG }, _> = Example { x: 0 };
    |                  ^                     ^
 
-error[E0107]: wrong number of const arguments: expected 1, found 0
+error[E0107]: wrong number of const arguments for `test_1`: expected 1, found 0
   --> $DIR/invalid-enum.rs:21:3
    |
+LL | pub fn test_1<const CF: CompileFlag>() {}
+   | -------------------------------------- defined here, with 1 const parameter: `CF`
+...
 LL |   test_1::<CompileFlag::A>();
-   |   ^^^^^^^^^^^^^^^^^^^^^^^^ expected 1 const argument
+   |   ^^^^^^   -------------- found no const argument in here
+   |
+help: add missing const argument:
+   |
+LL |   test_1::<CompileFlag::A, CF>();
+   |                          ^^^^
 
-error[E0107]: wrong number of type arguments: expected 0, found 1
-  --> $DIR/invalid-enum.rs:21:12
+error[E0107]: wrong number of type arguments for `test_1`: expected 0, found 1
+  --> $DIR/invalid-enum.rs:21:3
    |
+LL | pub fn test_1<const CF: CompileFlag>() {}
+   | -------------------------------------- defined here, with 0 type parameters
+...
 LL |   test_1::<CompileFlag::A>();
-   |            ^^^^^^^^^^^^^^ unexpected type argument
+   |   ^^^^^^
    |
-help: If this generic argument was intended as a const parameter, try surrounding it with braces:
+help: consider removing this type argument
+   |
+LL |   test_1::<>();
+   |           --
```

Comment:
The resulting "zero case" needs to be explicitly handled, so that the used span includes the `::<` and the closing `>` to remove the whole thing.

Reply:
done 🎉